db/hints/sync_point.cc

```diff
@@ -17,13 +17,15 @@
 #include "idl/hinted_handoff.dist.hh"
 #include "idl/hinted_handoff.dist.impl.hh"
 #include "utils/base64.hh"
+#include "utils/xx_hasher.hh"
 
 namespace db {
 namespace hints {
 
-// Format V1 (encoded in base64):
-//   uint8_t 0x01 - version of format
+// Format V1/V2 (encoded in base64):
+//   uint8_t 0x01/0x02 - version of format
 //   sync_point_v1 - encoded using IMR
+//   uint64_t - checksum computed using the xxHash algorithm (present only in V2)
```

Comment:
The comment would be more readable if "Format V2" had its own section IMO.

Reply:
Agree. I've also added a new comment:
```cpp
// Sync points can be encoded in two formats: V1 and V2. V2 extends V1 by adding
// a checksum. Currently, we use the V2 format, but sync points encoded in the V1
// format still can be safely decoded.
```
I think it's nice too understand this V1/V2 concept without having to analyze the code.